pkg/cmd/cli/cmd/importer/dockercompose.go

```diff
@@ -0,0 +1,182 @@
+package importer
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/kubectl"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	configcmd "github.com/openshift/origin/pkg/config/cmd"
+	"github.com/openshift/origin/pkg/generate/app"
+	appcmd "github.com/openshift/origin/pkg/generate/app/cmd"
+	"github.com/openshift/origin/pkg/generate/dockercompose"
+)
+
+const (
+	dockerComposeLong = `
+Import a Docker Compose file as OpenShift objects
```

Comment:
It will have different flags than process - we don't want compose specific
flags to overlap with process specific flags.

On Apr 29, 2016, at 3:23 PM, Ben Parees <notifications@github.com> wrote:

In pkg/cmd/cli/cmd/importer/dockercompose.go
<https://github.com/openshift/origin/pull/7690#discussion_r61632142>:

> +	"k8s.io/kubernetes/pkg/kubectl"
> +	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
> +	"k8s.io/kubernetes/pkg/runtime"
> +
> +	"github.com/openshift/origin/pkg/client"
> +	cmdutil "github.com/openshift/origin/pkg/cmd/util"
> +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> +	configcmd "github.com/openshift/origin/pkg/config/cmd"
> +	"github.com/openshift/origin/pkg/generate/app"
> +	appcmd "github.com/openshift/origin/pkg/generate/app/cmd"
> +	"github.com/openshift/origin/pkg/generate/dockercompose"
> +)
> +
> +const (
> +	dockerComposeLong = `
> +Import a Docker Compose file as OpenShift objects

it's not creating a new template unless you specify --as-template. The
default behavior is to create objects, which means you're processing a
template, it's just that the template syntax is Compose instead of
OpenShift.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/7690/files/b85cc800bdcb1e7db8f8c37dd3e85c21b5023f0a#r61632142>